src/main/java/impl/expression/Lexer.java

```diff
@@ -0,0 +1,59 @@
+package impl.expression;
+
+import api.expression.ParseException;
+import impl.expression.lexis.*;
+import impl.expression.lexis.Number;
+
+public class Lexer {
+
+    private final String input;
+    private int currentPosition;
+
+    public Lexer(String input) {
+        this.input = input;
+    }
+
+    public Lexeme getNextToken() throws ParseException {
+        if (currentPosition >= input.length()) {
+            return new Lexeme(LexemeType.EOE);
+        }
+        char symbol = input.charAt(currentPosition++);
+        switch (symbol) {
+            case '\t': //fallthrough
```

Comment:
Do you forget to add whitespace check?

Reply:
Umm, no, I've just thought that it'd be not that elegant with most of the meaningful code in `default` branch... 
Anyway, I can just do that, OR, I've had an inspiration and implemented some functional pattern-matching... if you want to have a look, I can push that. spoiler: 
```
        matcher = new Matcher<Character, Lexeme, LexerException>();
        matcher.match(Character::isWhitespace, s -> getNextToken());
        matcher.match(c -> c.equals('+'), s -> new Operator(OperatorType.PLUS));
        matcher.match(c -> c.equals('-'), s -> new Operator(OperatorType.MINUS));
        matcher.match(Character::isDigit, s -> new Number(consumeNumber(s)));
        matcher.orThrow(LexerException.class, "Improperly formatted input");
```
With only 

```
        char symbol = input.charAt(currentPosition++);
        return matcher.produce(symbol);
```
As actual iterating

Reply:
I like it more than checking for specific whitespace characters)

Reply:
So, `isWhitespace` two-line fix? =(

Reply:
I didn't get it again :(

Reply:
Do you prefer just the `isWhitespace` fix or overkill, I've mentioned?

Reply:
It doesn't matter for me :) I agree, it’s not cool that Java doesn't know how to do such checks, but you definitely need to check for whitespace